File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int minSoFar = prices[0];

        for(int i=1; i<prices.length; i++){
            int profit = prices[i] - minSoFar;

            if(profit > maxProfit){
                maxProfit = profit;
            }

            if(prices[i] < minSoFar){
                minSoFar = prices[i];
            }
        }

        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        minSoFar = prices[0]
        profit = 0

        for i in range(1, len(prices)):
            profit = max(profit, prices[i]-minSoFar)
            minSoFar = min(minSoFar, prices[i])

        return profit